### PR TITLE
[FIX] Unique domain checks

### LINKED ISSUE #4382: Domain: Prevent constructing domain with non-unique variable names
**Is your feature request related to a problem? Please describe.**
We have an agreement to construct domains with unique variable names, but there are no checks for that in the code.

**Describe the solution you'd like**
Raise an error when constructing a domain with duplicated variable names.

**Describe alternatives you've considered**
Create unique variable names under the hood when constructing the domain.

### PR DESCRIPTION
**Pull request by AndrejaKovacic**
##### Issue

Fixes  #4382

##### Includes
- [X] Code changes
- [x] Tests

### CODE REVIEW

**Comment by PrimozGodec** [comment]
Tests started to fail due to the new scikit release. Hope this PR fixes it https://github.com/biolab/orange3/pull/4768

File: Orange/widgets/unsupervised/owpca.py [hunk 1]
```diff
@@ -290,18 +292,33 @@ def commit(self):
...
+            d = self.data.domain
+
+            names = [var.name for var in chain(d.attributes, d.class_vars, d.metas) if var]
+            proposed = [var.name for var in transformed.domain.attributes[:self.ncomponents]]
+            unique = get_unique_names(names, proposed)
+
+            transformed_attributes = []
+            for var, u in zip(transformed.domain.attributes[:self.ncomponents], unique):
+                if var.name != u:
+                    transformed_attributes.append(var.copy(u))
```
[the remark this answers is not in the document]
Comment (AndrejaKovacic) [reply Orange/widgets/unsupervised/owpca.py]: The only way for this whole thing to make sense is if I looked up on the list that PCA is supposedly faulty, wrote some code which was then never actually executed and then finally tested the whole thing, observing how nicely it works.

**Comment by AndrejaKovacic** [comment]
Regarding Feature Constructor: variables are renamed one by one, immediately when created.  Change of name is visible in gui immediately. 
Renaming all the vars would result in issues with references in compute value for newly created vars. Renaming only new ones with get_unique_names has a problem if new names are not unique.

File: Orange/widgets/unsupervised/owpca.py [hunk 2]
```diff
@@ -292,33 +290,18 @@ def commit(self):
...
             transformed = transformed.from_table(domain, transformed)
             # prevent caching new features by defining compute_value
-            comp_names = [a.name for a in self._pca.orig_domain.attributes]
-            meta_var = get_unique_names(comp_names, 'component')
-
             dom = Domain(
                 [ContinuousVariable(a.name, compute_value=lambda _: None)
                  for a in self._pca.orig_domain.attributes],
-                metas=[StringVariable(name=meta_var)])
+                metas=[StringVariable(name='component')])
```
Comment (janezd) [inline Orange/widgets/unsupervised/owpca.py]: I fear you reverted too much.

Outputs "Data" and "Transformed data" don't need checking (as far as I see) but the third output, Components, still does. If apply PCA to data that contains a variable named "component", I get a table with two variables named "component".

**Pull request merged**